File: frontend/src/types/quant-item.ts
/**
 * 量化项目实体接口
 */
export interface QuantItem {
  id: number;
  name: string;
  category: string;
  description?: string;
  min_score: number;
  max_score: number;
  default_score?: number;
  default_reason?: string;
  weight: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

/**
 * 创建量化项目负载接口
 */
export interface CreateQuantItemPayload {
  name: string;
  category: string;
  description?: string;
  min_score: number;
  max_score: number;
  default_score?: number;
  default_reason?: string;
  weight?: number;
  is_active?: boolean;
}

/**
 * 更新量化项目负载接口
 */
export interface UpdateQuantItemPayload {
  name?: string;
  category?: string;
  description?: string;
  min_score?: number;
  max_score?: number;
  default_score?: number;
  default_reason?: string;
  weight?: number;
  is_active?: boolean;
}

/**
 * 量化项目过滤参数接口
 */
export interface QuantItemFilter {
  name?: string;
  category?: string | string[];
  is_active?: boolean | string;
}

/**
 * 量化项目分类接口
 */
export interface QuantItemCategory {
  id: number;
  name: string;
  description?: string;
  order?: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

/**
 * 创建量化项目分类负载接口
 */
export interface CreateQuantItemCategoryPayload {
  name: string;
  description?: string;
  order?: number;
  is_active?: boolean;
}

/**
 * 更新量化项目分类负载接口
 */
export interface UpdateQuantItemCategoryPayload {
  name?: string;
  description?: string;
  order?: number;
  is_active?: boolean;
} 

File: backend/app/models/quant_item.py
from sqlalchemy import Boolean, Column, Integer, String, Text, Numeric, DateTime, func, ForeignKey
from sqlalchemy.orm import relationship

from app.db.base import Base

class QuantItem(Base):
    __tablename__ = "quant_items"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False, comment="名称")
    description = Column(Text, nullable=True, comment="描述")
    min_score = Column(Numeric(5, 2), nullable=False, comment="最小分数")
    max_score = Column(Numeric(5, 2), nullable=False, comment="最大分数")
    default_score = Column(Numeric(5, 2), nullable=True, comment="默认分数")
    default_reason = Column(Text, nullable=True, comment="默认原因")
    weight = Column(Numeric(5, 2), default=1.0, nullable=False, comment="权重")
    category = Column(String(50), nullable=False, comment="类别")
    category_id = Column(Integer, ForeignKey("quant_item_categories.id"), nullable=True, comment="分类ID")
    is_active = Column(Boolean, default=True, nullable=False, comment="是否激活")
    created_at = Column(DateTime, default=func.now(), nullable=False, comment="创建时间")
    updated_at = Column(DateTime, default=func.now(), onupdate=func.now(), nullable=False, comment="更新时间")

    # 关系
    quant_records = relationship("QuantRecord", back_populates="item")
    category_rel = relationship("QuantItemCategory", back_populates="items")
